app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,51 @@
+class CommentsController < ApplicationController
+  before_action :authenticate_user!, except: [:index, :show]
+  before_action :find_post
+  before_action :find_comment, only: [:edit, :update, :destroy]
+
+  def create
+    @comment        = Comment.new comment_params
+    @comment.post   = @post
```

Comment:
Cute lil protip™ (Gabe):

We can initialize a comment with the id for `@post` by doing

```ruby
@comment = @post.comments.new(comment_params)
```